src/components/ScreenStack.tsx

```diff
@@ -7,6 +7,10 @@ import ScreenStackNativeComponent from '../fabric/ScreenStackNativeComponent';
 const NativeScreenStack: React.ComponentType<ScreenStackProps> =
   ScreenStackNativeComponent as any;
 
+function isFabric() {
+  return 'nativeFabricUIManager' in global;
+}
```

Comment:
IIRC this should be reliably available during the very first render, so we're safe on that front (the value won't differ on first & consecutive renders).